models/modules/diffusion_generator.py

```diff
@@ -155,12 +167,19 @@ def restoration_ddpm(
                 ref=ref,
                 guidance_scale=guidance_scale,
             )
-
+            if frame != 0:
+                mask = rearrange(mask, "b f c h w -> (b f) c h w")
             if mask is not None:
                 temp_mask = torch.clamp(mask, min=0.0, max=1.0)
                 y_t = y_0 * (1.0 - temp_mask) + temp_mask * y_t
             if i % sample_inter == 0:
                 ret_arr = torch.cat([ret_arr, y_t], dim=0)
+            if frame != 0:
```

Comment:
Maybe a dedicated function with all transforms for video input instead of having complicated code spread in different locations here ?